sed/core/processor.py

```diff
@@ -59,18 +59,25 @@ def __init__(
         self._coordinates: Dict[Any, Any] = {}
         self.axis: Dict[Any, Any] = {}
         self._attributes = MetaHandler(meta=metadata)
+
+        loader_name = self._config["core"]["loader"]
+        self.loader = get_loader(
+            loader_name=loader_name,
+            config=self._config,
+        )
+
         self.ec = EnergyCalibrator(
+            loader=self.loader,
```

Comment:
wouldn't it make sense to not load dataframe seperately for the calibrator? Giving it an input of a dataframe could work I believe. Right now, a dependency on the loader module has been created on calibrator module.

Reply:
With the code as it is, the energy calibrator needs to access the files directly to extract the bias voltage. This can be changed in the future, but in this PR I wanted to concentrate on one thing.
The reason for handling the energy calibration data seperately in the energycalibrator module is that these are separate data from the ones loaded into the processor. One could off course also handle the loading/binning there...